File: frontend/src/api/admin/branding.ts
import { apiRequest, API_BASE, getAccessToken } from '@/api/client'

export interface BrandingResponse {
  brand_name: string
  logo_url: string | null
  favicon_url: string | null
  primary_color: string
  secondary_color: string
  background_color: string
  font_family: string
  custom_css: string | null
  privacy_policy_url: string | null
  terms_of_service_url: string | null
  personal_data_url: string | null
  refund_policy_url: string | null
}

export interface PublicBrandingResponse extends BrandingResponse {
  news_enabled: boolean
  referral_enabled: boolean
  devices_enabled: boolean
  support_link: string | null
}

export interface BrandingUpdateRequest {
  brand_name?: string
  logo_url?: string
  favicon_url?: string
  primary_color?: string
  secondary_color?: string
  background_color?: string
  font_family?: string
  custom_css?: string
  privacy_policy_url?: string
  terms_of_service_url?: string
  personal_data_url?: string
  refund_policy_url?: string
}

export function getLegalContent(url: string): Promise<{ content: string }> {
  return fetch(`${API_BASE}/legal/content?url=${encodeURIComponent(url)}`).then(r => {
    if (!r.ok) throw new Error('Failed to load document')
    return r.json()
  })
}

export interface FeaturesResponse {
  news_enabled: boolean
  referral_enabled: boolean
  devices_enabled: boolean
}

export interface FeaturesUpdateRequest {
  news_enabled?: boolean
  referral_enabled?: boolean
  devices_enabled?: boolean
}

export function getPublicBranding(): Promise<PublicBrandingResponse> {
  return fetch(`${API_BASE}/config/branding`).then(r => r.json())
}

export function getAdminBranding(): Promise<BrandingResponse> {
  return apiRequest<BrandingResponse>('/admin/branding')
}

export function patchBranding(body: BrandingUpdateRequest): Promise<BrandingResponse> {
  return apiRequest<BrandingResponse>('/admin/branding', {
    method: 'PATCH',
    body: JSON.stringify(body),
  })
}

export async function uploadLogo(file: File): Promise<BrandingResponse> {
  const form = new FormData()
  form.append('file', file)
  const resp = await fetch(`${API_BASE}/admin/branding/logo`, {
    method: 'POST',
    headers: {
      Authorization: `Bearer ${getAccessToken()}`,
    },
    credentials: 'include',
    body: form,
  })
  if (!resp.ok) {
    const body = await resp.json().catch(() => ({ detail: 'Upload failed' }))
    throw new Error(body.detail ?? 'Upload failed')
  }
  return resp.json()
}

export function getAdminFeatures(): Promise<FeaturesResponse> {
  return apiRequest<FeaturesResponse>('/admin/features')
}

export function patchFeatures(body: FeaturesUpdateRequest): Promise<FeaturesResponse> {
  return apiRequest<FeaturesResponse>('/admin/features', {
    method: 'PATCH',
    body: JSON.stringify(body),
  })
}

File: web/schemas/admin/branding.py
from pydantic import BaseModel
from typing import Optional


class BrandingResponse(BaseModel):
    brand_name: str
    logo_url: Optional[str]
    favicon_url: Optional[str]
    primary_color: str
    secondary_color: str
    background_color: str
    font_family: str
    custom_css: Optional[str]
    privacy_policy_url: Optional[str] = None
    terms_of_service_url: Optional[str] = None
    personal_data_url: Optional[str] = None
    refund_policy_url: Optional[str] = None

    model_config = {"from_attributes": True}


class BrandingUpdateRequest(BaseModel):
    brand_name: Optional[str] = None
    logo_url: Optional[str] = None
    favicon_url: Optional[str] = None
    primary_color: Optional[str] = None
    secondary_color: Optional[str] = None
    background_color: Optional[str] = None
    font_family: Optional[str] = None
    custom_css: Optional[str] = None
    privacy_policy_url: Optional[str] = None
    terms_of_service_url: Optional[str] = None
    personal_data_url: Optional[str] = None
    refund_policy_url: Optional[str] = None


class PublicBrandingResponse(BrandingResponse):
    news_enabled: bool
    referral_enabled: bool
    devices_enabled: bool
    support_link: Optional[str] = None
    chat_widget_provider: Optional[str] = None
    chat_widget_id: Optional[str] = None
